spec/lib/common/connection-spec.js

```diff
@@ -5,112 +6,255 @@
+                        if (self.subject.connected === true) {
+                            return expect(self.subject.exchange('amqp')).to.be.rejectedWith(Error);
+                        }
+                    });
+                });
+
+                it('should publish exchange if name is present ', function () {
+                    var self = this;
+                    amqp.createConnection = sandbox.spy(function () {
+                        return ({
+                            'on': function (a, callback) {
+                                callback(null, {});
+                            },
+                            'exchange': function (a, b, callback){
+                                callback('exchange', {});
+                            },
+                            'exchanges':
+                                'on.test'
+                        });
+                    });
+                    return this.subject.start().then(function () {
+                        if (self.subject.connected === true) {
+                            var name = 2;
+                            return expect(self.subject.exchange(name)).to.be.ok;
+                        }
+                    });
                 });
             });
-        });
 
-        describe('stop', function () {
-            it('should fulfill on success', function () {
-                var self = this;
+            describe('stop', function () {
+                it('should close connection if on', function () {
+                    var self = this;
+                    amqp.createConnection = sandbox.spy(function () {
+                        return ({
+                            'on': function (a, callback) {
+                                if(a==='ready' || a=== 'close')
+                                {
+                                    callback(null, {});
+                                }
+                            }
+                        });
+                    });
+                    return this.subject.start().then(function () {
+                        return expect(self.subject.stop()).to.be.ok;
+                    });
+                });
 
-                return this.subject.start().then(function () {
-                    return self.subject.stop();
+                it('should disconnect connection', function (done) {
+                    var self = this;
+                    amqp.createConnection = sandbox.spy(function () {
+                        return ({
+                            'on': function (a, callback) {
+                                if(a === 'ready'){
+                                    callback(null, {});
+                                }
+                            },
+                            'disconnect':function(){done();},
+                        });
+                    });
+                    return this.subject.start().then(function () {
+                        return expect(self.subject.stop()).to.be.ok;
```

Comment:
Does `stop()` return a Promise? If so, we'll want to assert `return expect(self.subject.stop()).to.be.fulfilled` because I think `.to.be.ok` will pass even for an error object since it just detects a truthy/falsey condition.